server/index.js

```diff
@@ -98,13 +94,53 @@ app.post("/:username/interests", async (req, res) => {
                 interests: newInterests
             },
         })
-        console.log(res.json(updatedUser))
         res.json(updatedUser)
     } catch (error) {
         res.status(500)
     }
 })
 
+app.get("/get-hobbies", async (req, res) => {
+    try {
+        const hobbies = await prisma.hobby.findMany()
+        res.json(hobbies)
+    } catch (error) {
+        res.status(500)
+    }
+})
+
+app.get("/:username/get-interests", async (req, res) => {
+    const { username } = req.params
+    try {
+        const user = await prisma.user.findUnique({
+            where: { username: username },
+            select: { interests: true },
+        })
+        if (!user) {
+            return res.status(404)
+        }
+        res.json(user.interests)
+    } catch (error) {
+        console.error('Error fetching interests')
+        res.status(500)
+    }
+})
+
+app.post("/:username/update-hobby/:hobbyId", async (req, res) => {
```

Comment:
The username is in the route URL. Is this what you want? Technically anybody can type this into the browser. You should have a POST message body where you pass in an authentication token from the frontend, and then check that token here in this route.